enf-reducer/promote-common-constraints.metta

```diff
@@ -0,0 +1,68 @@
+;(: promoteCommonConstraints (-> Expression Expression))
+(= (promoteCommonConstraints $expr)
+    (let* 
+        (
+            (($literals $children) (getGsetAndChildren $expr)) 
+            ($childrenLiteral   (collapse (getGuardSet (superpose $children) ())))
+            ($shared (findCommon $childrenLiteral))
+            ($common (findIntersection $literals $shared))   
+            ($x 2)      
+        )
+        (if (== $common ())
+            (() $expr)
+            (let*
+                ( 
+                ($updatedChildren (removeCommonLiterals $common $children))
+                ($updatedLiterals (removeElement $common $literals))
+                ($newExp (concatTuple $updatedLiterals $updatedChildren ))
+                ($res (cons-atom OR $newExp))              
+                )
+                ($common $res)
```

Comment:
Any particular reason why you are returning `$common` in addition to the `$res`?

Reply:
`$common` needs to be added to the guard set of the parent of **POA** ,so it is returned along with the updated expression.